src/core/thread/address_resolver.cpp

```diff
@@ -243,6 +243,20 @@ void AddressResolver::UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAddre
         ExitNow();
     }
 
+    if (aApplyOptimization)
```

Comment:
Yes, also considered this option. not sure but if this way, `AddCacheEntry()` would have some binding with the `OT_ERROR_NOT_FOUND` retval of `UpdateCacheEntry()`, otherwise as a independent method, it need to iterate through the entries to ensure there is no existing entry.
Thoughts?

Reply:
no strong opinion on this. updated as suggestions. thx